### ISSUE TITLE
File monitoring

### ISSUE THREAD

**Issue by bigtfromaz** [jlesage/docker-crashplan#2]
This may be more of a question than an issue.  I installed your build a few days ago.  I adopted a machine that was running Windows and resynced everything and it all caught up.  Nice job by the way.  The issue I noticed is that file monitoring does not seem to work.  After a few days I noticed none of my new files had shown up in the restore tree.  When CrashPlan Pro ran on my Windows Server, a change to a file caused the pending file count to be incremented within a few minutes.  I think this is because CrashPlan registers for file monitoring and is called when files are touched.

I worked around the problem for now by changing the "Verify Selection" option for the backup to every 2 hours.  

Is the problem due to the fact that the Linux version of CrashPlan you installed does not monitor files?  Or is it that we need to do something in the Docker container to activate the capability?

**Comment by jlesage** [comment]
Hi,
File monitoring should definitely work.  Maybe your inotify limit is not enough high?  You can look at CrashPlan logs and search for a message containing 'Unable to add watch for path'.
Make sure to check the [Troubleshooting](https://github.com/jlesage/docker-crashplan#troubleshooting) section of the documentation for more details.

**Comment by bigtfromaz** [comment]
Hi,

Well…I have to say it’s been a long time since I was on the wrong side when closing out an issue with an RTFM.

I read your instructions and somehow I totally missed your  warning about inotify.  I set inotify a healthy number above my file count and all is good.

Sorry for the confusion.

Cheers,

BigTFromAZ

**Comment by jlesage** [comment]
No problem!

**Comment by bigtfromaz** [comment]
All is well but I have some additional information regarding Synology and also have a question.  This seems like the best place to put it.  

I deployed this container to a Synology NAS.  In the CODE42 notes regarding inotify [**here**](https://support.code42.com/CrashPlan/4/Troubleshooting/Linux_real-time_file_watching_errors), CrashPlan suggests changing /etc/sysctl.conf .  I would like to note that on a Synology the file name being used is /etc.defaults/sysctl.conf .  

Also, I am very new to Docker containers so this may be a dumb question.  I seems to me that Docker is a stripped down virtualization host.  Should I be making the inotify change somewhere in the container as well or is the real kernel being used all the way down to the CrashPlan app?  I am guessing the latter because I haven't seen a kernel being loaded anywhere.  Of course being a newbie I might have missed it.

Thanks!

BigT

By the way, reading your work has been enlightening.  I think we'll be deploying our own company's apps using Docker going forward.  I am going to try pulling your package into a Windows Docker host today to see how it goes.

**Comment by jlesage** [comment]
A Docker container is more like a shroot than a VM.  No virtualisation takes place and everything inside the container runs natively on the host.
Thus, inotify limit is set on the host and nothing needs to be done on the container side.

**Comment by bigtfromaz** [comment]
That's what I surmised.  The confusion came when I installed Docker for Windows and it required Hyper-V to be running.  I have since learned that Microsoft is doing that to provide an extra layer of isolation.  More importantly, it's to provide a base on which to deploy containers running Linux-based images.  They are doing this in a pre-release version which relies on a VM name MobyLinuxVM, which is a LinuxKit implementation.  

As a learning experience I am going to spin up a flavor of your CrashPlan Pro image based on MobyLinux to see if I can get it all to work.
